core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -556,6 +562,42 @@ public Object readGenericValue() throws IOException {
         return null;
     }
 
+    /**
+     * Reads a {@link NamedWriteable} from the current stream, by first reading its name and then looking for
+     * the corresponding entry in the registry by name, so that the proper object can be read and returned.
+     */
+    <C> C readNamedWriteable(String category) throws IOException {
+        String name = readString();
+        NamedWriteable<C> namedWriteable = namedWriteableRegistry.getPrototype(category, name);
+        return namedWriteable.readFrom(this);
```

Comment:
Should we make this method throw an exception instead and have the ability to provide a namedWriteableRegistry in a subclass of FilterStreamInput which would first try to read using its own registry and otherwise delegate to its parent stream input?